jpo-ode-plugins/src/test/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverterTest.java

```diff
@@ -482,4 +462,240 @@ void ensureComplianceWithJ2735Revision2024_containsMultipleOldFields() {
     });
   }
 
-}
+  @Test
+  void transformNodeLL_LLTypeSpecified() throws JsonUtilsException, TravelerMessageFromHumanToAsnConverter.InvalidNodeLatLonOffsetException {
+    // prepare
+    ObjectNode node = JsonNodeFactory.instance.objectNode();
+    node.put("nodeLong", "0.0031024");
+    node.put("nodeLat", "0.0014506");
+    node.put("delta", "node-LL3");
+
+    // execute
+    ObjectNode result = TravelerMessageFromHumanToAsnConverter.transformNodeLL(node);
+
+    // verify
+    String expectedJson = "{\"delta\":{\"node-LL3\":{\"lat\":14506,\"lon\":31024}}}";
+    ObjectNode expected = JsonUtils.toObjectNode(expectedJson);
+
+    Assertions.assertEquals(expected.toString(), result.toString());
+  }
+
+  @Test
+  void transformNodeLL_LLTypeNotSpecified() throws JsonUtilsException, TravelerMessageFromHumanToAsnConverter.InvalidNodeLatLonOffsetException {
+    // prepare
+    ObjectNode node = JsonNodeFactory.instance.objectNode();
+    node.put("nodeLong", "-0.0008192");
+    node.put("nodeLat", "-0.0013123");
+    node.put("delta", "node-LL");
+
+    // execute
+    ObjectNode result = TravelerMessageFromHumanToAsnConverter.transformNodeLL(node);
+
+    // verify
+    String expectedJson = "{\"delta\":{\"node-LL3\":{\"lat\":-13123,\"lon\":-8192}}}";
+    ObjectNode expected = JsonUtils.toObjectNode(expectedJson);
+
+    Assertions.assertEquals(expected.toString(), result.toString());
+  }
+
+  long LL_1_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL1_LIMIT * -1;
+  long LL_2_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL2_LIMIT * -1;
+  long LL_3_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL3_LIMIT * -1;
+  long LL_4_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL4_LIMIT * -1;
+  long LL_5_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL5_LIMIT * -1;
+  long LL_6_MINIMUM_MICRO_DEGREES = TravelerMessageFromHumanToAsnConverter.NODE_LL6_LIMIT * -1;
+
+  @Test
+  void determineNodeOffsetPointLLType_WhenLatLonWithinLL1Constraints_ShouldReturnDetermineNodeLLType1()
+      throws TravelerMessageFromHumanToAsnConverter.InvalidNodeLatLonOffsetException {
+    long transformedLat = -1500L; // within LL1 constraints
+    long transformedLong = -1500L; // within LL1 constraints
+    String selection = TravelerMessageFromHumanToAsnConverter.determineNodeOffsetPointLLType(transformedLat, transformedLong);
+
+    Assertions.assertEquals("node-LL1", selection,
```

Comment:
suggestion(non-blocking): use the constants you defined instead of a magic string here. This helps users understand more easily where the expected value is defined. If these values were at risk of changing (which I don't think they are) then the tests would be easier to maintain by using the constants.

This suggestion applies to all the tests below, too

```suggestion
    Assertions.assertEquals(TravelerMessageFromHumanToAsnConverter.NODE_LL1, selection,
```

Reply:
Refactored to use constants!